scheds/rust/scx_chaos/src/bpf/main.bpf.c

```diff
@@ -457,15 +463,38 @@ void BPF_STRUCT_OPS(chaos_runnable, struct task_struct *p, u64 enq_flags)
 	return p2dq_runnable_impl(p, enq_flags);
 }
 
+void BPF_STRUCT_OPS(chaos_running, struct task_struct *p)
+{
+	struct chaos_task_ctx *taskc;
+
+	if (!(taskc = lookup_create_chaos_task_ctx(p))) {
+		scx_bpf_error("failed to lookup task context in enqueue");
+		return;
+	}
+
+	p2dq_running_impl(p);
```

Comment:
Nte that `p2dq_running_impl` can also set frequency so the logic comes afterwards. Maybe it's a good idea to have a bool to disable it?

Reply:
If you think the places this is called in p2dq will stay quite static let's leave it like this (and make sure the reset works fine too). Maybe a comment about the ordering would be good too? I want to stay as close to p2dq behaviour as possible when it doesn't hit chaos

Reply:
Sounds good, I had a comment and lost it in a rebase.